### PR TITLE
Feature/top visible view controller

### PR DESCRIPTION
**Pull request by petropavel13**

### CODE REVIEW

File: LeadKit/LeadKit/Extensions/Observable/Observable+ToastErrorLogging.swift [hunk 1]
```diff
@@ -29,11 +29,12 @@ public extension Observable {
...
     /// - Returns: The source sequence with the side-effecting behavior applied.
     func showErrorsInToastInDebugMode() -> Observable<Observable.E> {
-        return `do`(onError: { (error) in
-            #if DEBUG
-                UIApplication.shared.keyWindow?.makeToast(error.localizedDescription)
-            #endif
-        })
+        return observeOn(MainScheduler.instance)
+            .do(onError: { (error) in
+                #if DEBUG
```
Comment (iON1k) [inline LeadKit/LeadKit/Extensions/Observable/Observable+ToastErrorLogging.swift]: Maybe move DEBUG macro to method body?

Comment (iON1k) [inline LeadKit/LeadKit/Extensions/Observable/Observable+ToastErrorLogging.swift]: And I think that observeOn(MainScheduler.instance) is not good idea.

How about this solution?
dispatch_async(GlobalMainQueue) {
        UIApplication.shared.keyWindow?.makeToast(error.localizedDescription)
}

Comment (petropavel13) [inline LeadKit/LeadKit/Extensions/Observable/Observable+ToastErrorLogging.swift]: >Maybe move DEBUG macro to showErrorsInToastInDebugMode method body?

what for?

>And I think that observeOn(MainScheduler.instance) is not good idea.

why?

File: LeadKit/LeadKit/Extensions/UserDefaults/UserDefaults+MappableDataTypes.swift [hunk 2]
```diff
@@ -166,12 +157,7 @@ public extension Reactive where Base: UserDefaults {
     ///
     /// - returns: Observable of specified model type.
     func object<T>(forKey key: String, defaultValue: T) -> Observable<T> where T: ImmutableMappable {
-        return Observable.create { observer in
-            observer.onNext(self.base.object(forKey: key, defaultValue: defaultValue))
-            observer.onCompleted()
-
-            return Disposables.create()
-        }
+        return Observable.just(self.base.object(forKey: key, defaultValue: defaultValue))
```
Comment (iON1k) [inline LeadKit/LeadKit/Extensions/UserDefaults/UserDefaults+MappableDataTypes.swift]: deferredJust

File: LeadKit/LeadKit/Extensions/UserDefaults/UserDefaults+MappableDataTypes.swift [hunk 3]
```diff
@@ -202,12 +179,7 @@ public extension Reactive where Base: UserDefaults {
     ///
     /// - returns: Observable of specified array type.
     func object<T>(forKey key: String, defaultValue: [T]) -> Observable<[T]> where T: ImmutableMappable {
-        return Observable.create { observer in
-            observer.onNext(self.base.object(forKey: key, defaultValue: defaultValue))
-            observer.onCompleted()
-
-            return Disposables.create()
-        }
+        return Observable.just(self.base.object(forKey: key, defaultValue: defaultValue))
```
Comment (iON1k) [inline LeadKit/LeadKit/Extensions/UserDefaults/UserDefaults+MappableDataTypes.swift]: deferredJust

**Pull request merged**